Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -387,6 +387,27 @@ Gfx::IntRect TextEditor::visible_text_rect_in_inner_coordinates() const
     };
 }
 
+template<typename DrawCallback, typename MoveCallback>
+void draw_text_with_tabs_helper(const Utf32View& visual_line_text, DrawCallback draw, MoveCallback move)
```

Comment:
Please use a different name for the `MoveCallback` since `move` clashes with `std::move()` which is basically a pseudo-language-feature used consistently throughout the project.

Also, instead of using a callback for this, how about passing the `Gfx::IntRect` to this helper as a mutable reference? Then we could `move_by()` the rect from here without needing a second lambda.